esp/services/ws_dfu/ws_dfuService.hpp

```diff
@@ -30,6 +30,81 @@
 #include "fvrelate.hpp"
 #include "dadfs.hpp"
 
+class CThorNodeGroup: public CInterface
+{
+    CDateTime timeCached;
+    StringAttr groupName;
+    unsigned keyhash;
+    unsigned nodeCount;
+    bool replicateOutputs;
+
+public:
+    IMPLEMENT_IINTERFACE;
+    CThorNodeGroup(const char* _groupName, unsigned _nodeCount, bool _replicateOutputs)
+        : groupName(_groupName), nodeCount(_nodeCount), replicateOutputs(_replicateOutputs)
+    {
+        keyhash = hashc((const byte *)groupName.get(),groupName.length(),0);
+        timeCached.setNow();
+    }
+
+    inline unsigned queryHash() { return keyhash; }
+    inline const char *queryGroupName() { return groupName.get(); }
+
+    inline bool queryNodeCount() { return nodeCount; };
+    inline bool queryReplicateOutputs() { return replicateOutputs; };
+    inline bool queryCanReplicate() { return replicateOutputs && (nodeCount > 1); };
+    inline bool checkTimeOut(unsigned timeOutMinutes)
+    {
+        CDateTime timeLine;
```

Comment:
@richardkchapman what do you think about using CDateTime?  It is more expensive than msTick() - will it be significant here?  
We could possibly do with a CDateTime.setNow(delta) function, which would be more efficient, or a function to compare a CDateTime with a delta from now.  E.g., timeCached.compareNow(-timeOutMinutes).  Fairly minor though.